packages/typespec-azure-core/lib/models.tsp

```diff
@@ -400,3 +400,12 @@ model ArmResourceIdentifierAllowedResource {
    */
   scopes?: ArmResourceDeploymentScope[];
 }
+
+@doc("Used in file part of multipart request body")
+model MultiPartFile extends File {
```

Comment:
not sure this name makes much sense in describing that everything is required. The fact that it sent over multipart(which is already the main use case of the standard `File`) doesn't imply requireness of filename or contentType

Reply:
I ever thought about some names:
- `FileWithRequiredMeta`: it seems too long
- `FileDetails`: not specific
Any idea?

Reply:
`FileWithRequiredMetadata` seems ok to me, but yeah it's definitely a balance between being too long and being descriptive enough.

`RequiredFile`? tbh I don't love that

Reply:
`ExplicitFile`?
`FileWithRequiredMetadata` seems ok as well

Reply:
Updated with `FileWithRequiredMetadata`